Lib/test/test_codecs.py

```diff
@@ -1020,6 +1020,7 @@ def test_errors(self):
             (b'a+////,+IKw-b', 'a\uffff\ufffd\u20acb'),
             (b'a+IKw-b\xff', 'a\u20acb\ufffd'),
             (b'a+IKw\xffb', 'a\u20ac\ufffdb'),
+            (b'a+@b', 'a\ufffdb'),
```

Comment:
This test will fail. Is that you want?

Reply:
Sorry, I put the comment in the wrong line. I think that the test will fail on line 1029 `self.assertEqual(raw.decode('utf-7', 'replace'), expected)`

Reply:
Well, you're mistaken.

Reply:
Please add also tests for `b'a+'` and `b'a+@'`. It may be need to change the code for proper handling these cases.